### PR TITLE
Use webpack for CSS, Sass and Stylus as well as JavaScript (resolves #27)

### LINKED ISSUE #27: Add CSS/SCSS/Stylus processing
## Is your feature request related to a problem?

The ability to compile SCSS or Stylus and process CSS using PostCSS would be a valuable addition to this starter kit.

## Describe the solution you'd like

The best approach would be to use Eleventy to process stylesheets directly as seen in [Eleventastic](http://github.com/maxboeck/eleventastic) and [inclusive-design/idrc](https://github.com/inclusive-design/idrc/).

## Describe alternative solutions you've considered

Not applicable.

## Additional context or notes

CSS, Stylus, or SCSS should all be supported out of the box. [Autoprefixer](https://github.com/postcss/autoprefixer) should be run as will as a minification routine.

### PR DESCRIPTION
**Pull request by greatislander**
This PR consolidates CSS, Sass, Stylus and JavaScript processing using [webpack](https://webpack.js.org) (specifically, the [Laravel Mix](https://laravel-mix.com) wrapper). Documentation can be found [here](https://github.com/greatislander/fluidic-11ty/blob/webpack/src/assets/README.md).

### CODE REVIEW

File: package.json
Comment (jobara) [inline package.json:43]: Would be preferable to link to an actual release. I spoke with @the-t-in-rtf in the [fluid-work channel today](http://irc-logs.fluidproject.org/%23fluid-work/%23fluid-work.2020-10-15.log) about it. We are just waiting on @amb26 to sign off on https://github.com/fluid-project/infusion/pull/1013. After that we can merge in https://github.com/fluid-project/eslint-config-fluid/pull/8 and https://github.com/fluid-project/fluid-grunt-lint-all/pull/15 (needs a new release of the eslint-config-fluid) and create new releases for both respectively.

Comment (greatislander) [reply package.json:43]: Agreed.

File: src/assets/README.md
Comment (jobara) [inline src/assets/README.md:11]: It's nice to have direct references but using line numbers here will be brittle, as future changes may affect them, and it may not be obvious for those making code changes that they need to come back here to updated the references. Is there a way to identify where to look for the code without actually having to link to the specific line numbers?

Comment (greatislander) [reply src/assets/README.md:11]: @jobara Once this is committed I could link to the file at the specific commit reference.

Comment (jobara) [reply src/assets/README.md:11]: Is there some alternative wording for "configuration details can be found here" to use as the link text. Maybe rewording the sentence like "See [PostCSS Plugins configuration](https://laravel-mix.com/docs/5.0/css-preprocessors#postcss-plugins) for information on how to add and configure additional [plugins](https://github.com/postcss/postcss/blob/master/docs/plugins.md)"

File: src/assets/README.md
Comment (jobara) [inline src/assets/README.md:15]: Is there documentation that we can link to for the asset manifest?

File: webpack.mix.js
Comment (jobara) [inline webpack.mix.js:36]: I think you may have forgotten to uncomment line 33 and 36

Comment (greatislander) [reply webpack.mix.js:36]: I didn't forget—because the input files are all called `app.{css|scss|styl}` it would cause a filename collision if we tried to compile them all to the same output path. The idea here is to show what the configuration would look like for CSS, Sass, and Stylus, not to have active configurations for all three at once.

File: src/assets/README.md
Comment (greatislander) [reply src/assets/README.md:15]: Not really. I guess I could put an example manifest into the README.

File: src/assets/README.md
Comment (greatislander) [reply src/assets/README.md]: Resolved in e97afd9.

File: src/assets/README.md
Comment (greatislander) [reply src/assets/README.md]: Resolved in e97afd9.

File: webpack.mix.js
Comment (jobara) [reply webpack.mix.js:36]: Okay so this will only process a single file called `app.{css|scss|styl}` and by default that is `app.css` If I want instead to use an scss file I'd have to add the scss file and modify this file to process the new scss file instead . 

I don't think this is clear in the documentation, and the automated checking was a bit nicer before. 

If you want to say have your styles separated into separate files, you'd need to import them into `app.{css|scss|styl}` to be processed and usable by the site? Is that correct?

Would it be possible to go through the assets/styles directory and just process each file as needed and output to distribute/assets/styles. I think the only changes you'd need to do would be to loop through the style files and run the appropriate processor/compiler and to use different names for your example style files. In this way the project using this setup can use whichever file name and type they want without having to touch the configuration file. I guess a downside is that if you want separate files but have them imported into a single file, you'll need to add those in a separate directory.

Another thought. What if you want different styles for different pages? For example your news posts vs the landing page.

Comment (greatislander) [reply webpack.mix.js]: > Okay so this will only process a single file called `app.{css|scss|styl}` and by default that is `app.css` If I want instead to use an scss file I'd have to add the scss file and modify this file to process the new scss file instead .
>
> I don't think this is clear in the documentation, and the automated checking was a bit nicer before.

I included all three file types in `asset/styles`, the idea being that you could delete the ones you don't need and uncomment the configuration line you needed, deleting or commenting out the rest. I've changed things so that the configuration automatically processes all files at the top level of the `src/assets/styles` directory, regardless of file type. This necessitated removing the example Sass and Stylus files to avoid output file conflicts but I've clarified that in the documentation.

> If you want to say have your styles separated into separate files, you'd need to import them into `app.{css|scss|styl}` to be processed and usable by the site? Is that correct?

That's correct. It really depends on how you want to structure your CSS/Sass/Stylus.

> Would it be possible to go through the assets/styles directory and just process each file as needed and output to distribute/assets/styles. I think the only changes you'd need to do would be to loop through the style files and run the appropriate processor/compiler and to use different names for your example style files. In this way the project using this setup can use whichever file name and type they want without having to touch the configuration file. I guess a downside is that if you want separate files but have them imported into a single file, you'll need to add those in a separate directory.

That's not a downside so much as if you want to structure your CSS/Sass/Stylus with imports you just have to do that anyway :) I've made this change so now all the filetypes are handled automatically.

> Another thought. What if you want different styles for different pages? For example your news posts vs the landing page.

Multiple CSS/Sass/Stylus files are supported now so this should "just work". However, the files are still loaded on all pages so I need to look into that.

File: src/_includes/partials/stylesheets.njk
Comment (jobara) [inline src/_includes/partials/stylesheets.njk:6]: @greatislander I'm running into a strange issue here. I was testing out the processing of multiple style files. I used both the same file type and different file types. What appears to be happening is that some of the files are being missed when expanding the partial. The files all appear in the dist directory but references aren't all showing up in the `<link>` tags in the html page. I'm not even sure how it is missing files as it doesn't depend on the exact name, nor the alphabetical order. It might also just use 2 of them, as that's the most I've gotten to appear so far.

File: package.json
Comment (BlueSlug) [inline package.json:53]: `resolve-url-loader`, `sass-loader` and `stylus-loader` are all listed as dependencies but are not used. Should they be added to the webpack configuration to provide an example or a baseline config? Would this be unnecessary due to the way Laravel Mix does its thing?

**Review by BlueSlug — COMMENTED** [review]
Looking pretty good! I had an issue where the CSS and UIO weren't being served the first time I ran the server, but on every subsequent restart, it seemed to work fine. Hopefully it was just a caching issue :)

File: package.json
Comment (greatislander) [reply package.json:53]: They're used internally by Laravel Mix so they don't need to be imported or used in an example config in any way.

File: src/assets/styles/app.css
Comment (greatislander) [reply src/assets/styles/app.css]: Resolved in 406a1a507ef53f467137ed6e88b9cc557e400d00.

File: src/_includes/partials/stylesheets.njk
Comment (greatislander) [reply src/_includes/partials/stylesheets.njk]: I believe this is resolved now, @jobara (see 6e23937).

**Comment by greatislander** [comment]
@BlueSlug @jhung @jobara I did a little research into an initialization script that would set up the desired processor but it felt like a dead end so I just simplified the configuration to include just CSS by default and expanded the [docs](https://github.com/greatislander/fluidic-11ty/blob/webpack/src/assets/README.md). Let me know what you think!

**Review by BlueSlug — APPROVED** [review]
Looks great :)

`webpack.mix.js` is much shorter and easier to read at a glance, so this should be better for newer developers. I really like the +/- lines in the docs, as well!

I'm good with this as-is, but for super extra points, I wonder if it's worth documenting the previous code from the Webpack config and the regex search in the template. What do you think?

**Comment by greatislander** [comment]
@BlueSlug Yeah, I think that could be worth doing— maybe I can add a section at the end of the docs called "Advanced Techniques" or something. Cheers!

**Pull request merged**
